src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -94,6 +107,31 @@ public Resource Merge(Resource other)
             return new Resource(newAttributes);
         }
 
+        /// <summary>
+        /// Returns a new <see cref="Resource"/> with added attributes from telemetry sdk and the <see cref="OtelEnvResourceDetector"/>.
+        /// </summary>
+        /// <returns><see cref="Resource"/>.</returns>
+        internal Resource GetResourceWithDefaultAttributes()
+        {
+            return this.Merge(TelemetryResource).Merge(new OtelEnvResourceDetector().Detect());
+        }
+
+        /// <summary>
+        /// Returns a new <see cref="Resource"/> with added attributes from resource detectors in the order of the list.
+        /// </summary>
+        /// <param name="detectors">A list of <see cref="IResourceDetector"/>.</param>
+        /// <returns><see cref="Resource"/>.</returns>
+        internal Resource GetResourceFromDetectors(List<IResourceDetector> detectors)
+        {
+            var resource = this;
+            foreach (IResourceDetector detector in detectors)
+            {
+                resource = resource.Merge(detector.Detect());
```

Comment:
Not a blocker, but once Detector is public, anyone can write own detector, and it throws exception, this whole method and possibly SDK itself would crash.

Reply:
okay for now, as this is internal.

Reply:
Okay. Yeah. We can circle back to this in subsequent PRs.